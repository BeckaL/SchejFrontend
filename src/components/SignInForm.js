import React, { Component } from 'react'
import axios from 'axios'
import { setAuthStorage } from '../actions/authentication'

export default class SignInForm extends Component {
  constructor(props) {
    super(props)

    this.state = {
      email: "",
      password: "",
      authenticationToken: localStorage['authenticationToken']
    }
}

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  handleSubmit = event => {
    event.preventDefault()
    this.createUserRequest()
  }
  
  createUserRequest = () => {
    const self = this
    axios.post(
      'http://localhost:3001/api/v1/sign_in',
      {
        email: this.state.email,
        password: this.state.password
      })
      .then(response => {
        setAuthStorage(response.data)
        self.props.history.push('/shifts')
        console.log(response)
      })
      .catch(error => {
        if (error.response.status === 422) {
          alert('Wrong email and password')
        } else {
          console.log(error)
        }
      })
    }

    render() {
      return (
        <form className='sign-in-form' onSubmit={this.handleSubmit}>
          <input id='sign-in-email-entry' className='sign-in-entry' type='text' name='email' placeholder='Email' onChange={this.handleChange} /><br />
          <input id='sign-in-password-entry' className='sign-in-entry' type='password' name='password' placeholder='Password' onChange={this.handleChange} /><br />
          <button id='sign-in-submit' className='custom-button'>Submit</button>
        </form>
      )
    }
    }
