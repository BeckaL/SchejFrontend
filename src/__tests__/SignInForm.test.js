import React from 'react'
import { create } from "react-test-renderer"
import { mount, shallow } from 'enzyme'
import SignInForm from '../components/SignInForm'
import axios from "axios"

jest.mock('axios')

describe('sign in form', () => {
  var component
  var wrapper

  beforeAll(() => {
    component = create(<SignInForm history = {[]}/>)
    wrapper = shallow(<SignInForm history = {[]}/>)
  })

  beforeEach(() => {
    window.localStorage.clear()
  })

  test("it matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  test("it posts to the correct API when sign in credentials are correct", () => {
    const response = {
      data: {}
    }

    axios.post.mockResolvedValue(response)

    const form = component.root.findByType('form')

    const fakeEvent = { preventDefault: () => {} }

    form.props.onSubmit(fakeEvent)

    expect(axios.post).toHaveBeenCalledTimes(1)
    expect(axios.post).toHaveBeenCalledWith(
      'http://localhost:3001/api/v1/sign_in', {
        email: "",
        password: ""
      }
    )
  })

  test('It has a disabled submit button by default', () => {
    const button = wrapper.find('#sign-in-submit')

    expect(button.props('disabled')).toBeTruthy()
  })
})
