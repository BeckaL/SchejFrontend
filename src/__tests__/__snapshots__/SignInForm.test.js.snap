// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign in form it matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignInForm
    history={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <IndexTitle />,
        <form
          className="sign-in-form"
          onSubmit={[Function]}
        >
          <input
            className="sign-in-entry"
            id="sign-in-email-entry"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <br />
          <input
            className="sign-in-entry"
            id="sign-in-password-entry"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <br />
          <button
            className="custom-button"
            disabled={true}
            id="sign-in-submit"
          >
            Submit
          </button>
        </form>,
        <button
          className="custom-button"
          id="sign-up-button"
          onClick={[Function]}
        >
          Sign Up
        </button>,
      ],
      "id": "sign-in-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="sign-in-entry"
              id="sign-in-email-entry"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />,
            <br />,
            <input
              className="sign-in-entry"
              id="sign-in-password-entry"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />,
            <br />,
            <button
              className="custom-button"
              disabled={true}
              id="sign-in-submit"
            >
              Submit
            </button>,
          ],
          "className": "sign-in-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sign-in-entry",
              "id": "sign-in-email-entry",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sign-in-entry",
              "id": "sign-in-password-entry",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "custom-button",
              "disabled": true,
              "id": "sign-in-submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "className": "custom-button",
          "id": "sign-up-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <IndexTitle />,
          <form
            className="sign-in-form"
            onSubmit={[Function]}
          >
            <input
              className="sign-in-entry"
              id="sign-in-email-entry"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />
            <br />
            <input
              className="sign-in-entry"
              id="sign-in-password-entry"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
            <br />
            <button
              className="custom-button"
              disabled={true}
              id="sign-in-submit"
            >
              Submit
            </button>
          </form>,
          <button
            className="custom-button"
            id="sign-up-button"
            onClick={[Function]}
          >
            Sign Up
          </button>,
        ],
        "id": "sign-in-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="sign-in-entry"
                id="sign-in-email-entry"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
              />,
              <br />,
              <input
                className="sign-in-entry"
                id="sign-in-password-entry"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
              />,
              <br />,
              <button
                className="custom-button"
                disabled={true}
                id="sign-in-submit"
              >
                Submit
              </button>,
            ],
            "className": "sign-in-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "sign-in-entry",
                "id": "sign-in-email-entry",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "sign-in-entry",
                "id": "sign-in-password-entry",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "custom-button",
                "disabled": true,
                "id": "sign-in-submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "custom-button",
            "id": "sign-up-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
